sympy/solvers/bivariate.py

```diff
@@ -205,11 +321,33 @@ def _solve_lambert(f, symbol, gens):
     if not lamcheck:
         raise NotImplementedError()
 
+    even_degrees = [i for i in _power_list(lhs, symbol).get(symbol, []) \
```

Comment:
In this section here it looks like you parse for even degrees, if you find them you do a symbol replacement and then you *check to see if you made the replacement*. I would suggest eliminating the checking for even degrees and just do so by doing the subs/check part of what you have. 

Note the following behavior:
```python
>>> (x**3).subs(x**2,y)
x**3
>>> (x**4).subs(x**2,y)
y**2
```
In other words: if there are even powers the replacement will happen, else it won't.
You can make the replacement even more targeted as:
```python
>>> (1+x+x**2+x**4+x**3).replace(lambda i:i.is_Pow and i.base == x and i.exp%2==0, lambda i: y**i.exp)
x**3 + x + y**4 + y**2 + 1
```

So if you want to know if you only have even degrees in variable `symbol` then do 
```
tlhs = lhs.subs(symbol**2, t)
if symbol not in tlhs and tlhs.has(t):
    # all powers were even
    tpow = [i for i in tlhs.atoms(Pow) if i.base == t]
    if not tpow:
        # symbol**2 -> t
    elif len(tpow) == 1:
        # symbol**even -> t**(even/2)
    else:
        # there were multiple even powers
```

Reply:
> You can make the replacement even more targeted as:
(1+x+x**2+x**4+x**3).replace(lambda i:i.is_Pow and i.base == x and i.exp%2==0, lambda i: y**i.exp)

This code is a hit from the sweet spot of the bat!!. It does all the process in a single step of finding all even powers and replacing them.

> So if you want to know if you only have even degrees in variable symbol then do

We are not looking for expressions only just to have even powers. The expression can have any even power along with odd power. What we have to do is just with even powers, this doesn't mean we restrict the expression to have odd degrees.

>  I would suggest eliminating the checking for even degrees and just do so by doing the subs/check part of what you have.

I will try to fulfill this thing.